File: questik/app/src/main/java/ru/rinet/questik/data/room/model/MetricsDao.java
package ru.rinet.questik.data.room.model;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Transaction;
import androidx.room.Update;

import java.util.List;

@Dao
public interface MetricsDao {
    @Transaction
    @Query("SELECT _id, name FROM metrics")
    List<MetricsWithJob> getMetricsWithJob();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Metrics metrics);

    @Update
    void update(Metrics metrics);

    @Delete
    void delete(Metrics metrics);
}

File: app/src/main/java/ru/rinet/questik/repo/local/room/dao/MetricsDao.kt
package ru.rinet.questik.repo.local.room.dao

import androidx.paging.PagingSource
import androidx.room.*
import ru.rinet.questik.repo.local.room.entity.MetricsEntity

@Dao
interface MetricsDao {

    @Query("SELECT COUNT(id) FROM metrics")
    fun getMetricsCount(): Int


    @Query("SELECT * FROM metrics")
    fun getAllMetrics(): PagingSource<Int, MetricsEntity>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllMetrics(items:List<MetricsEntity>)

    @Query("DELETE FROM metrics")
    fun clearMetrics()

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertMetrics(metrics: MetricsEntity)

    @Update
    fun updateMetrics(metrics: MetricsEntity)

    @Delete
    fun deleteMetrics(metrics: MetricsEntity)

    @Query("SELECT * FROM metrics WHERE id == :id")
    fun getMetricsById(id: Int): List<MetricsEntity>

    @Query("SELECT * FROM metrics")
    fun getMetrics(): List<MetricsEntity>
}